### PR TITLE
Weekly test for legacy features

### PR DESCRIPTION
**Pull request by yhaliaw**
Applicable spec: <link>

### Overview

Move some test to run weekly, rather than on every PR.

The `Integration test with juju 2.9` test status check will needed to be disabled with this PR.

### Rationale

Development will shift to OpenStack integration runners. Focus will be on juju 3+.

The local LXD runners and juju 2.9 tests will not need to be ran every PR.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by cbartz — DISMISSED** [review]

**Review by arturo-seijas — DISMISSED** [review]

File: .github/workflows/e2e_test.yaml [hunk 1]
```diff
@@ -1,470 +1,32 @@
...
+    secrets: inherit
     with:
-      runner-tag: "pr-${{ needs.run-id.outputs.run-id }}${{ github.run_attempt}}"
-      runner-virt-type: "lxd"
+      juju-channel: 3.2/stable
+      pre-run-script: scripts/setup-lxd.sh
+      provider: lxd
+      test-tox-env: integration-juju3.2
+      modules: '["test_e2e"]'
+      extra-arguments: "-m openstack --openstack-flavor-name=builder-cpu4-ram8-disk50 --http-proxy=http://squid.internal:3128 --https-proxy=http://squid.internal:3128 --no-proxy=keystone.ps6.canonical.com,glance.ps6.canonical.com,nova.ps6.canonical.com,neutron.ps6.canonical.com"
```
Comment (cbartz) [inline .github/workflows/e2e_test.yaml:19]: This is information about internal infrastructure, should it really be public or should it be placed in a secret?

File: .github/workflows/e2e_test.yaml [hunk 2]
```diff
@@ -1,466 +1,32 @@
...
     runs-on: ubuntu-latest
     needs:
-      - deploy-e2e-test-runner
+      - openstack-integration-end-to-end-test
     if: always() && !cancelled()
     timeout-minutes: 5
     steps:
       - run: |
-          [ '${{ needs.deploy-e2e-test-runner.result }}' = 'success' ] || (echo deploy-e2e-test-runner failed && false)
+          [ '${{ needs.openstack-integration-end-to-end-test.result }}' = 'success' ] || (echo openstack-integration-end-to-end-test failed && false)
```
Comment (yanksyoon) [inline .github/workflows/e2e_test.yaml:32]: Small missing new-line :)

File: .github/workflows/integration_test.yaml [hunk 3]
```diff
@@ -35,7 +29,7 @@ jobs:
       pre-run-script: scripts/setup-lxd.sh
       provider: lxd
       test-tox-env: integration-juju3.2
-      modules: '["test_charm_metrics_failure", "test_charm_metrics_success", "test_charm_fork_repo", "test_charm_runner", "test_e2e"]'
+      modules: '["test_charm_metrics_failure", "test_charm_metrics_success", "test_charm_fork_repo", "test_charm_runner"]'
       extra-arguments: "-m openstack --openstack-flavor-name=builder-cpu4-ram8-disk50 --http-proxy=http://squid.internal:3128 --https-proxy=http://squid.internal:3128 --no-proxy=keystone.ps6.canonical.com,glance.ps6.canonical.com,nova.ps6.canonical.com,neutron.ps6.canonical.com"
       self-hosted-runner: true
       self-hosted-runner-label: stg-private-endpoint
```
Comment (yanksyoon) [inline .github/workflows/integration_test.yaml:35]: Missing newline here as well :D

File: .github/workflows/scheduled_integration_test.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  integration-tests:
+    name: Integration test with juju 3.1
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    secrets: inherit
+    with:
+      juju-channel: 3.1/stable
+      pre-run-script: scripts/pre-integration-test.sh
+      provider: lxd
+      test-tox-env: integration-juju3.1
+      modules: '["test_charm_base_image", "test_charm_fork_repo", "test_charm_no_runner", "test_charm_scheduled_events", "test_charm_lxd_runner", "test_charm_runner", "test_charm_metrics_success", "test_charm_metrics_failure", "test_self_hosted_runner", "test_charm_with_proxy", "test_charm_with_juju_storage", "test_debug_ssh", "test_charm_upgrade"]'
```
Comment (yanksyoon) [inline .github/workflows/scheduled_integration_test.yaml:32]: Missing newline here as well :D

**Review by yanksyoon — DISMISSED** [review]
LGTM!

**Review by yanksyoon — DISMISSED** [review]

**Review by yanksyoon — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

**Pull request merged**